lib/generators.nix

```diff
@@ -144,90 +148,93 @@ rec {
+          # "<δ:${concatStringsSep "," (builtins.attrNames v)}>"
+        else "{ " + libStr.concatStringsSep " " (libAttr.mapAttrsToList
+          (name: value:
+            "${toPretty args name} = ${toPretty args value};") v) + " }"
+      else if isFunction v then
+        let
+          fna = lib.functionArgs v;
           showFnas = concatStringsSep "," (libAttr.mapAttrsToList
-                       (name: hasDefVal: if hasDefVal then "(${name})" else name)
-                       fna);
-      in if fna == {}    then "<λ>"
-                         else "<λ:{${showFnas}}>"
-    else abort "generators.toPretty: should never happen (v = ${v})";
+            (name: hasDefVal: if hasDefVal then "(${name})" else name)
+          fna);
+        in
+          if fna == { } then "<λ>"
+          else "<λ:{${showFnas}}>"
+      else abort "generators.toPretty: should never happen (v = ${v})";
 
   # PLIST handling
-  toPlist = {}: v: let
-    isFloat = builtins.isFloat or (x: false);
-    expr = ind: x:  with builtins;
-      if x == null  then "" else
-      if isBool x   then bool ind x else
-      if isInt x    then int ind x else
-      if isString x then str ind x else
-      if isList x   then list ind x else
-      if isAttrs x  then attrs ind x else
-      if isFloat x  then float ind x else
-      abort "generators.toPlist: should never happen (v = ${v})";
-
-    literal = ind: x: ind + x;
-
-    bool = ind: x: literal ind  (if x then "<true/>" else "<false/>");
-    int = ind: x: literal ind "<integer>${toString x}</integer>";
-    str = ind: x: literal ind "<string>${x}</string>";
-    key = ind: x: literal ind "<key>${x}</key>";
-    float = ind: x: literal ind "<real>${toString x}</real>";
-
-    indent = ind: expr "\t${ind}";
-
-    item = ind: libStr.concatMapStringsSep "\n" (indent ind);
-
-    list = ind: x: libStr.concatStringsSep "\n" [
-      (literal ind "<array>")
-      (item ind x)
-      (literal ind "</array>")
-    ];
-
-    attrs = ind: x: libStr.concatStringsSep "\n" [
-      (literal ind "<dict>")
-      (attr ind x)
-      (literal ind "</dict>")
-    ];
-
-    attr = let attrFilter = name: value: name != "_module" && value != null;
-    in ind: x: libStr.concatStringsSep "\n" (lib.flatten (lib.mapAttrsToList
-      (name: value: lib.optional (attrFilter name value) [
-      (key "\t${ind}" name)
-      (expr "\t${ind}" value)
-    ]) x));
-
-  in ''<?xml version="1.0" encoding="UTF-8"?>
+  toPlist = {}: v:
+    let
+      isFloat = builtins.isFloat or (x: false);
+      expr = ind: x: with builtins;
+        if x == null then "" else
+          if isBool x then bool ind x else
+            if isInt x then int ind x else
+              if isString x then str ind x else
+                if isList x then list ind x else
+                  if isAttrs x then attrs ind x else
+                    if isFloat x then float ind x else
+                      abort "generators.toPlist: should never happen (v = ${v})";
```

Comment:
Indentation is wrong, but got fixed in the newest commit of `nixpkgs-fmt`'s `nixpkgs-repository` branch